c10/xpu/XPUDeviceProp.h

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+#include <c10/xpu/XPUMacros.h>
+#include <sycl/sycl.hpp>
+
+namespace c10::xpu {
+
+struct C10_XPU_API DeviceProp {
+  // Returns the device name of this SYCL device.
+  sycl::info::device::name::return_type device_name;
+
+  // Returns the device type associated with the device.
+  sycl::info::device::device_type::return_type device_type;
```

Comment:
Not sure if syncl::info::device is a namespace or not, but I wonder if anything can be done to avoid typing the same over and over again
```suggestion
  using syncl_device = syncl::info::device;
  sycl_device::name::return_type device_name;

  // Returns the device type associated with the device.
  sycl_device::device_type::return_type device_type;
```

Reply:
sycl::info::device is a namespace, here we use a MARCO to avoid typing too much same code. Do you think it is OK?